src/server/scripts/Northrend/Naxxramas/boss_razuvious.cpp

```diff
@@ -179,20 +296,65 @@ class boss_razuvious_minion : public CreatureScript
 public:
     boss_razuvious_minion() : CreatureScript("boss_razuvious_minion") { }
 
-    CreatureAI* GetAI(Creature* pCreature) const override
+    CreatureAI* GetAI(Creature* creature) const override
     {
-        return GetNaxxramasAI<boss_razuvious_minionAI>(pCreature);
+        return GetNaxxramasAI<boss_razuvious_minionAI>(creature);
     }
 
     struct boss_razuvious_minionAI : public ScriptedAI
     {
-        explicit boss_razuvious_minionAI(Creature* c) : ScriptedAI(c) { }
-
-        EventMap events;
+        explicit boss_razuvious_minionAI(Creature* creature) : ScriptedAI(creature) { }
 
         void Reset() override
         {
-            events.Reset();
+            scheduler.CancelAll();
+            ScheduleAttackDummy();
+        }
+
+        void ScheduleAttackDummy()
+        {
+            me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_STATE_READY1H);
+            if (Creature* targetDummy = me->FindNearestCreature(NPC_TARGET_DUMMY, 10.0f))
+            {
+                me->SetFacingToObject(targetDummy);
+            }
+            scheduler.Schedule(6s, 9s, GROUP_OOC_RP, [this](TaskContext context)
+            {
+                me->HandleEmoteCommand(EMOTE_ONESHOT_ATTACK1H);
+                context.Repeat(6s, 9s);
+            });
+        }
+
+        void DoAction(int32 action) override
+        {
+            switch (action)
+            {
+                case ACTION_FACE_ME:
+                    scheduler.CancelGroup(GROUP_OOC_RP);
+                    me->SetSheath(SHEATH_STATE_UNARMED);
+                    me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_STATE_NONE);
+                    if (InstanceScript* instance = me->GetInstanceScript())
+                    {
+                        if (Creature* creature = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_RAZUVIOUS_BOSS)))
```

Comment:
same here, we could use instance->GetCreature()

Reply:
Yes, here the storage can be used since there is only one razzuvious spawned

Reply:
tried in d3bf91c7d9b444731715e120d8c2d6186e844243, failed to get Razuvious this way. The instancescript needs a refactor